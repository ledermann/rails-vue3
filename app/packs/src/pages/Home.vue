<template>
  <h1 class="mt-10 text-4xl font-extrabold tracking-tight text-white sm:text-5xl lg:text-6xl">
    Hello {{ name }}!
  </h1>
  <p class="mt-6 text-xl text-indigo-100 max-w-3xl">
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum ipsa eos harum pariatur asperiores accusantium qui facere eligendi aspernatur! Ea nobis quo quisquam facere. At, itaque. Repellendus eius ut neque?
  </p>

  <custom-dropdown
    class="mt-10"
    :items="dropdownItems"
    name="This is a Dropdown"
  />
</template>

<script>
import CustomDropdown from '@/components/CustomDropdown.vue'

const dropdownItems = [
  { name: 'My Account', href: '#my-account' },
  { name: 'Company', href: '#company' },
  { name: 'Team Members', href: '#team' },
  { name: 'Billing', href: '#billing' },
]

export default {
  name: 'Home',

  components: {
    CustomDropdown,
  },

  props: {
    name: {
      type: String,
      required: true,
    },
  },

  setup() {
    return {
      dropdownItems,
    }
  },
}
</script>
