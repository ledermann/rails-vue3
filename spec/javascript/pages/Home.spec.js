import { mount } from '@vue/test-utils'
import Home from '@/pages/Home.vue'

describe('Home', () => {
  test('renders name', () => {
    const wrapper = mount(Home, {
      propsData: {
        name: 'World',
      }
    })

    const title = wrapper.find('h1').element.textContent.trim()
    expect(title).toEqual('Hello World!')
  })

  test('matches snapshot', () => {
    const wrapper = mount(Home, {
      propsData: {
        name: 'World',
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
