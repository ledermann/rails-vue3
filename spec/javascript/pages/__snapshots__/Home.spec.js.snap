// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<h1 class="mt-10 text-4xl font-extrabold tracking-tight text-white sm:text-5xl lg:text-6xl"> Hello World! </h1>
<p class="mt-6 text-xl text-indigo-100 max-w-3xl"> Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum ipsa eos harum pariatur asperiores accusantium qui facere eligendi aspernatur! Ea nobis quo quisquam facere. At, itaque. Repellendus eius ut neque? </p>
<div class="relative inline-block text-left mt-10">
  <div><button id="headlessui-menu-button-3" type="button" aria-haspopup="true" class="inline-flex justify-center w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-sm font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-indigo-500">This is a Dropdown <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="-mr-1 ml-2 h-5 w-5">
        <path fill-rule="evenodd" d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z" clip-rule="evenodd"></path>
      </svg></button></div>
  <transition-stub>
    <!---->
  </transition-stub>
</div>
`;
